cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -277,53 +276,81 @@ public static String clientsCaCertSecretName(String cluster) {
     }
 
     public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
+        return fromCrd(kafkaAssembly, versions, null);
+    }
+
+    public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions, Storage oldStorage) {
```

Comment:
just an idea, instead of passing a specific thing like the `Storage` got from the STS in the KAO, could it be useful passing the STS itself for getting the old storage here? Maybe having the current STS instance could be useful in the future? Or maybe I am just overthinking it ...

Reply:
Do you see anything like that in the near future? TBH, I think it is better to keep this logic outside to make it easier to change and keep the `KafkaCluster` and `ZookeeperCluster` cleaner.